tbx/core/blocks.py

```diff
@@ -511,8 +537,38 @@ def clean(self, value):
         return struct_value
 
 
+class EventImageChooserBlock(ImageChooserBlock):
```

Comment:
Andy's suggestion for this was that it would be more maintainable to create a version of the ImageChooser that uses prefetch, but which fetches all the renditions for the images, so that it is more generic and can be used by multiple blocks.

Reply:
I wanted to go that way, but felt too much to prefetch all renditions given we have quite a few variations

Reply:
This thing is... wherever we are using an ImageBlock, that's curated content, so there are always going to be a limited number of uses per page. 

Let's be generous and say that the page uses 30 of these throughout it's content. And that the images has 15 renditions each.

- Fetching all renditions will mean a total of 450 rendition in-memory at one time
- Fetching only the selected renditions will mean a total of 90 renditions in-memory at one time

While smaller numbers is obviously more desirable, this isn't going to make a real-world difference for a page. A `Rendition` object uses a negligible amount of memory (it's just a class instance with it's raw field values stored in `__dict__`). And the fact that we're prefetching renditions at all is where we're most benefiting from a database perspective: If you look at the real difference (in terms of time and resources) in selecting 450 renditions from the DB instead of 90 - I'd be surprised if the difference were really that measurable.

Reply:
I don't disagree with you.

5fe9c152

Reply:
This looks good as far as I can tell - tested a few block types to see and all working as expected.